packages/yoastseo/src/morphology/dutch/stem.js

```diff
@@ -46,48 +61,32 @@ const removeSuffixFromFullForms = function( exceptionsRemoveSuffixFromFullForms,
  *
  * @param {string} word The word to check.
- * @param {Object} morphologyDataNLStemming The Dutch morphology data file.
+ * @param {Object} morphologyDataNL The Dutch morphology data file.
  * @returns {string|null} The stem or null if the word was not matched by any of the exception checks.
  */
-const checkOtherStemmingExceptions = function( word, morphologyDataNLStemming ) {
+const checkOtherStemmingExceptions = function( word, morphologyDataNL ) {
 	/*
 	 * Check whether the word is on an exception list of adjectives with stem ending in -rd. If it is, stem and
 	 * return the word here, instead of going through the regular stemmer.
  	 */
-	const wordAfterRdExceptionCheck = stemAdjectiveEndingInRd( word, morphologyDataNLStemming.stemExceptions.adjectivesEndInRD );
+	const wordAfterRdExceptionCheck = stemAdjectiveEndingInRd( word, morphologyDataNL.stemming.stemExceptions.adjectivesEndInRD );
 	if ( wordAfterRdExceptionCheck !== word ) {
 		return wordAfterRdExceptionCheck;
 	}
-	/* Checks whether the word is in the exception list of nouns with specific diminutive or plural suffixes that needs to be stemmed.
-	 * If it is return the stem here.
-	 */
-	const stemFromFullForm = removeSuffixFromFullForms( morphologyDataNLStemming.stemExceptions.removeSuffixFromFullForms, word );
-	if ( stemFromFullForm ) {
-		return stemFromFullForm;
-	}
-
 	/*
-	 * Checks whether the word is in the ending match sub-list of diminutives that need to be stemmed and that additionally need
-	 * to have the final vowel removed. If it is return the stem here. Example: zwemdiplomaatje -> zwemdiplomaa -> zwemdiploma
+	 * Checks whether the word is in the exception list of nouns or adjectives with specific suffixes that needs to be stemmed.
+	 * If it is return the stem here and run possible stem modification if it is required. e.g. modes -> mod -> mood
 	 */
-	const stemFromFullFormAndDeleteFinalVowelEndingMatch = removeSuffixFromFullForm(
-		morphologyDataNLStemming.stemExceptions.stemTjeAndOnePrecedingVowel.forms.endingMatch,
-		morphologyDataNLStemming.stemExceptions.stemTjeAndOnePrecedingVowel.suffix,
-		word
-	);
-
-	if ( stemFromFullFormAndDeleteFinalVowelEndingMatch ) {
-		return stemFromFullFormAndDeleteFinalVowelEndingMatch.slice( 0, -1 );
-	}
-
-	/*
-	 * Checks whether the word is in the exact match sub-list of diminutives that need to be stemmed and that additionally need
-	 * to have the final vowel removed. If it is return the stem here. Example: omaatje -> omaa -> oma
-	 */
-	if ( morphologyDataNLStemming.stemExceptions.stemTjeAndOnePrecedingVowel.forms.exactMatch.includes( word ) ) {
-		return word.slice( 0, -4 );
+	let stemFromFullForm = removeSuffixFromFullForms( morphologyDataNL, word );
+	if ( stemFromFullForm ) {
+		const checkIfDoublingVowelIsNeeded = isVowelDoublingAllowed( stemFromFullForm, morphologyDataNL.stemming.stemExceptions,
+			morphologyDataNL.verbs.compoundVerbsPrefixes );
+		if ( checkIfDoublingVowelIsNeeded ) {
```

Comment:
This can be simplified to:

```suggestion
		if ( isVowelDoublingAllowed( stemFromFullForm, morphologyDataNL.stemming.stemExceptions,
			morphologyDataNL.verbs.compoundVerbsPrefixes ) ) {
```